lib/server.js

```diff
@@ -361,122 +363,116 @@ function createServer(options, callback) {
-            log: options.log,
-            crc_mode: client_crc_mode
-        }, function (cErr, clients) {
-            if (cErr) {
-                throw new verror.VError(cErr, 'unable to create boray clients');
-            }
-
-            opts.clients = clients;
+        var labels = {
+            datacenter: options.datacenter,
+            server: options.server_uuid,
+            zonename: os.hostname(),
+            pid: process.pid
+        };
 
-            var labels = {
-                datacenter: options.datacenter,
-                server: options.server_uuid,
-                zonename: os.hostname(),
-                pid: process.pid
-            };
+        var collector = artedi.createCollector({
+            labels: labels
+        });
 
-            var collector = artedi.createCollector({
-                labels: labels
-            });
+        collector.gauge({
+            name: 'client_crc_mode',
+            help: 'The node-fast CRC compatibilty mode of the Fast client'
+        }).set(client_crc_mode);
+
+        collector.gauge({
+            name: 'server_crc_mode',
+            help: 'The node-fast CRC compatibilty mode of the Fast server'
+        }).set(server_crc_mode);
+
+        var socket = net.createServer({ 'allowHalfOpen': true });
+        var server = new fast.FastServer({
+            collector: collector,
+            log: log.child({ component: 'fast' }),
+            server: socket,
+            crc_mode: server_crc_mode
+        });
 
-            collector.gauge({
-                name: 'client_crc_mode',
-                help: 'The node-fast CRC compatibilty mode of the Fast client'
-            }).set(client_crc_mode);
-
-            collector.gauge({
-                name: 'server_crc_mode',
-                help: 'The node-fast CRC compatibilty mode of the Fast server'
-            }).set(server_crc_mode);
-
-            var socket = net.createServer({ 'allowHalfOpen': true });
-            var server = new fast.FastServer({
-                collector: collector,
-                log: log.child({ component: 'fast' }),
-                server: socket,
-                crc_mode: server_crc_mode
-            });
+        var methods = [
+            {
+                rpcmethod: 'getobjectlocation',
+                rpchandler: getObjectLocation(opts)
+            },
+            {
+                rpcmethod: 'getplacementdata',
+                rpchandler: getPlacementData(opts)
+            },
+            {
+                rpcmethod: 'getvnodes',
+                rpchandler: getVnodes(opts)
+            }
+        ];
 
-            var methods = [
-                { rpcmethod: 'getbucket', rpchandler: getBucket(opts) },
-                { rpcmethod: 'createbucket', rpchandler: createBucket(opts) },
-                { rpcmethod: 'deletebucket', rpchandler: deleteBucket(opts) },
-                { rpcmethod: 'listbuckets', rpchandler: listBuckets(opts) },
```

Comment:
I can see we're deleting this method here, but I can't see us delete the function later on in this change, and we seem to keep the DTrace probe too. Do these functions and DTrace probes still serve a purpose in this service, but we just don't want to expose them as RPCs?

The same applies to a few other of these methods (at least "listobjects" that I spotted).

Reply:
No purpose at all - I just hadn't updated this file as I should.  I'll make sure all of the functions and unneeded code is removed in a follow up commit on this PR. good catch.